### PR TITLE
Automate Grasping Reach

### PR DESCRIPTION
**Pull request by avagdumortesin**
Fixes https://github.com/foundryvtt/pf2e/issues/2322.

### CODE REVIEW

File: packs/data/feats.db/grasping-reach.json [hunk 1]
```diff
@@ -19,10 +19,42 @@
...
+                        "melee",
+                        "weapon:hands-held:2",
+                        "grasping-reach"
+                    ],
+                    "any": [
+                        "weapon:damage:die:faces:6",
+                        "weapon:damage:die:faces:8",
+                        "weapon:damage:die:faces:10",
+                        "weapon:damage:die:faces:12"
+                    ],
```
Comment (stwlam) [inline packs/data/feats.db/grasping-reach.json:50]: This isn't necessary

File: packs/data/feats.db/grasping-reach.json [hunk 2]
```diff
@@ -19,10 +19,42 @@
...
+                    "not": [
+                        "weapon:traits:reach"
+                    ]
+                },
+                "selector": "damage"
```
Comment (stwlam) [inline packs/data/feats.db/grasping-reach.json:55]:
```suggestion
                "selector": "strike-damage"
```

File: packs/data/feats.db/grasping-reach.json (hunk 1 above)
Comment (avagdumortesin) [reply packs/data/feats.db/grasping-reach.json:50]: Would you suggest using `"weapon:damage:die:faces:4"` in the `"not"` predicate instead? The feat requires the weapon have a damage die of d6 or larger.

Comment (stwlam) [reply packs/data/feats.db/grasping-reach.json:50]: Sure, but note also that the weapon must require two hands to be wielded, not merely that it must be held in two hands.

Comment (avagdumortesin) [reply packs/data/feats.db/grasping-reach.json:50]: Hmm... so exclude weapons with the Two-Hand Dx trait?

File: packs/data/feats.db/grasping-reach.json [hunk 3]
```diff
@@ -19,10 +19,42 @@
...
             "value": 1
         },
+        "location": null,
         "prerequisites": {
             "value": []
         },
-        "rules": [],
+        "rules": [
+            {
+                "domain": "all",
```
Comment (stwlam) [inline packs/data/feats.db/grasping-reach.json:28]:
```suggestion
                "domain": "strike-damage",
```

File: packs/data/feats.db/grasping-reach.json [hunk 4]
```diff
@@ -19,10 +19,42 @@
...
+            },
+            {
+                "key": "DamageDice",
+                "override": {
+                    "downgrade": true
+                },
+                "predicate": {
+                    "all": [
+                        "melee",
+                        "weapon:hands-held:2",
```
Comment (stwlam) [inline packs/data/feats.db/grasping-reach.json:42]:
```suggestion
                        "weapon:usage:hands:2",
```

Comment (avagdumortesin) [reply packs/data/feats.db/grasping-reach.json]: Added

File: packs/data/feats.db/grasping-reach.json (hunk 3 above)
Comment (avagdumortesin) [reply packs/data/feats.db/grasping-reach.json]: Updated

File: packs/data/feats.db/grasping-reach.json (hunk 1 above)
Comment (avagdumortesin) [reply packs/data/feats.db/grasping-reach.json]: Updated

File: packs/data/feats.db/grasping-reach.json [hunk 5]
```diff
@@ -19,10 +19,37 @@
...
+        "location": null,
         "prerequisites": {
             "value": []
         },
-        "rules": [],
+        "rules": [
+            {
+                "domain": "strike-damage",
+                "key": "RollOption",
+                "label": "Grasping Reach Grip Adjusted",
```
Comment (stwlam) [inline packs/data/feats.db/grasping-reach.json:30]:
```suggestion
```

File: packs/data/feats.db/grasping-reach.json [hunk 6]
```diff
@@ -19,10 +19,36 @@
...
+                "toggleable": true
+            },
+            {
+                "key": "DamageDice",
+                "override": {
+                    "downgrade": true
+                },
+                "predicate": {
+                    "all": [
+                        "melee",
```
Comment (stwlam) [inline packs/data/feats.db/grasping-reach.json:40]:
```suggestion
                        "weapon:melee",
```

**Pull request merged**